src/components/MainButton/MainButton.test.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import browser from 'webextension-polyfill';
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { buttons as Content } from '@src/content.json'
+import { SearchStatus, SearchStatusType } from '@src/lib/internal-types'
+import { ActionTypes, PlatformMessage } from '@src/platform-message';
+import { MainButton, SearchMessage } from './MainButton'
+
+describe('Render MainButton', () => {
+    it('should render a "stop search" button if search is started or warning', () => {
```

Comment:
This tests are amazing, IMHO it would be much cleaner if you will create a driver like we have in `App.test.tsx` or even use the same driver as we have there